lesson22/index.js

```diff
@@ -0,0 +1,257 @@
+    }
+};
+
+const createErrorMessage = text => {
+    const errorText = createElementWithClassName("p", "error-message");
+    errorText.textContent = text;
+    return errorText;
+};
+
+const renderTable = data => {
+    const table = createElementWithClassName("table", "table");
+    const titleKeys = Object.keys(tableTitlesData);
+    const titleNames = Object.values(tableTitlesData);
+    const tableHead = createTableHead(titleNames);
+    const tableBody = createTableBody(data, titleKeys);
+
+    parent.appendChild(table).appendChild(tableHead).after(tableBody);
+};
+
+const createTableHead = titlesData => {
+    const tableHead = document.createElement("thead");
+    const tr = createElementWithClassName("tr", "table__row");
+    const fragment = document.createDocumentFragment();
+    
+    for(let i = 0; i < titlesData.length; i++){
+        const th = createElementWithClassName("th", "table__title js-table-title");
+        th.textContent = titlesData[i];
+        fragment.appendChild(th);
+    }
+    tableHead.appendChild(tr).appendChild(fragment);
+
+    return tableHead;
+};
+
+const createTableBody = (data, keys) => {
+    const tableBody = document.createElement("tbody");
+    const fragment = document.createDocumentFragment();
+
+    for(let i = 0; i < data.length; i++){
+        const tr = createElementWithClassName("tr", "table__row js-table-row");
+        const td = createTableContents(data[i], keys);
+        fragment.appendChild(tr).appendChild(td);
+    }
+    tableBody.appendChild(fragment);
+
+    return tableBody;
+};
+
+const createTableContents = (data, keys) => {
+    const fragment = document.createDocumentFragment();
+
+    for(let i = 0; i < keys.length; i++){
+        const td = createElementWithClassName("td", "table__contents js-table-contents");
+        td.textContent = data[keys[i]];
+        fragment.appendChild(td);
+    }
+
+    return fragment;
+};
+
+const createSortButtons = () => {
+    const ul = createElementWithClassName("ul", "sort-buttons");
+    ul.id = "js-sort-buttons";
+
+    const buttonConfig = [
+        { src: "../img/icon-both.svg", alt: "昇順に並び替える", dataSet: Sort.Default},
+        { src: "../img/icon-asc.svg", alt: "降順に並び替える", dataSet: Sort.Asc},
+        { src: "../img/icon-desc.svg", alt: "順不同に並び替える", dataSet: Sort.Desc}
+    ]
+
+    const fragment = document.createDocumentFragment();
+    buttonConfig.forEach(item => {
+        const li = createElementWithClassName("li", "sort-buttons__item js-sort-button-item");
+        const button = createElementWithClassName("button", "sort-buttons__btn");
+        const img = createElementWithClassName("img", "sort-buttons__img");
+
+        li.dataset.buttonStatus = item.dataSet;
+        img.src = item.src;
+        img.alt = item.alt;
+        fragment.appendChild(li).appendChild(button).appendChild(img);
+    })
+    ul.appendChild(fragment);
+    return ul;
+};
+
+const renderSortButton = () => {
+    const sortTarget = ["ID", "年齢"]
```

Comment:
IMO
I thought it would be better if I could reference it from an already defined object.
```suggestion
      const sortTarget = [tableTitlesData.userId, tableTitlesData.age];
```

Reply:
You are so right!
I realized in this assignment that the reference from the object is weak.
Will learn to use them better in the future!
Thank you!
[9ba0cb8](https://github.com/haru-programming/JavaScript-lessons/pull/27/commits/9ba0cb8e5e617f72baba4125c043360869f3a317)